### ISSUE TITLE
formatter: CSS declaration whose name starts with a digit fails to parse, but Prettier formats it

### ISSUE THREAD

**Issue by miloag** [oxc-project/oxc#26290]
### What happens

Oxfmt rejects a declaration whose name starts with a digit. Prettier formats the same input unchanged, so this is a Prettier-compatibility gap rather than invalid input for the tools that consume it.

```css
@text {
  xl: 1.25rem, 1.75rem;
  2xl: 1.5rem, 2rem;
}
```

```
  x Syntax error: CSS rule is expected
   ,-[src/index.css:3:3]
 2 |   xl: 1.25rem, 1.75rem;
 3 |   2xl: 1.5rem, 2rem;
   :   ^^^
 4 | }
   `----
```

Prettier 3.9.6 (`--parser css`) formats it unchanged. PostCSS reads `2xl` as a word token, so every PostCSS-based tool accepts the declaration.

### Why it matters

A parse error fails the whole file, so `vp check` stops before lint and type-check run, and the only escape is `fmt.ignorePatterns`, which drops the file from formatting altogether.

CSS dialects built on PostCSS use this shape for token tables. The example is [Rainbow Index](https://github.com/rainbowindex/rainbowindex)'s `@text` directive, where the name is the suffix of a utility class (`text-2xl`), so it cannot be renamed.

### Related

Same class of gap as #26158 (fixed in #26194): a PostCSS-style nested configuration block.

### Versions

- oxfmt 0.66.0
- prettier 3.9.6 for the comparison
- macOS, Node 24
